source/ios/AdaptiveCards/ADCIOSFramework/ADCIOSFramework/ACRSeparator.mm

```diff
@@ -0,0 +1,173 @@
+//
+//  ACRSeparator.mm
+//  ADCIOSFramework
+//
+//  Copyright © 2017 Microsoft. All rights reserved.
+//
+
+#import "ACRSeparator.h"
+#import "HostConfig.h"
+#import "TextBlock.h"
+#import <UIKit/UIKit.h>
+
+using namespace AdaptiveCards;
+
+@implementation ACRSeparator
+{
+    CGFloat width;
+    CGFloat height;
+    CGFloat lineWidth;
+    UILayoutConstraintAxis axis;
+    long rgb;
+}
+
++ (void) renderSeparation:(std::shared_ptr<BaseCardElement> const &) elem
+             forSuperview:(UIView*) view
+           withHostConfig:(std::shared_ptr<HostConfig> const &) config
+{          
+    ACRSeparator* separator = nil;
+    if(SeparationStyle::None != elem->GetSeparationStyle() &&
+       [view isKindOfClass: [UIStackView class] ])
+    {
+        UIStackView* superview = (UIStackView*) view;
+        separator = [[ACRSeparator alloc] init];
+        SeparationConfig* separatorHstCnfig = 
+            [separator getSeparationConfig: elem
+                       withHostConfig: config];
+        if(separator && separatorHstCnfig)
```

Comment:
exception handling isn't very robust in objective-c. let me check about that, and add such at next iteration.

---
In reply to: [126748824](https://github.com/Microsoft/AdaptiveCards/pull/454#discussion_r126748824) [](ancestors = 126748824)